### PR TITLE
Task: Add ability to set timeout

### PR DESCRIPTION
**Pull request by JafarAbdi**
I have found that when one of the stages is struggling to find a solution the task can get stuck planning forever, so in this PR I exposed the setTimeout function and made the task fail if it exceeded the time

### CODE REVIEW

**Comment by mamoll** [comment]
This seems generally useful. While MTC heuristically guides the search for a solution, there is often still a combinatorial explosion of candidate solutions that MTC will consider. It is often not desirable to exhaustively try all candidate solutions.

**Review by rhaschke — APPROVED** [review]
Nice contribution. Computing the remaining time incrementally is somewhat hard to follow.
I replaced this with the difference between _current_ and _start_ time, which should be less than the timeout.
